costmodel/costmodel.go

```diff
@@ -2144,6 +2166,16 @@ func (cm *CostModel) costDataRange(cli prometheusClient.Client, clientset kubern
 			}
 		}
 
+		var podStatefulSets []string
+		if _, ok := podStatefulsetsMapping[nsKey]; ok {
```

Comment:
If `nsKey := c.Namespace + "," + c.ClusterID` how does this match with the `podStatefulsetsMapping` keys, which are `nsKey := ns + "," + ss + "," + clusterID`?

Reply:
podStatefulsetsMapping should be a map[string]map[string][]string where the key is two levels, [namespace,clusterid][podname]

Reply:
Yes, but I'm pointing out that there might be in issue here: https://github.com/kubecost/cost-model/blob/93f04c52cf43e32111747f30703a3e23769ea51f/costmodel/promparsers.go#L376

The type is still `map[string]map[string][]string`, but I think the top-level keys will not match.

Reply:
Perhaps that `nsKey` in the promparsers for StatefulSets should just be `ns,clusterId`, no?

Reply:
Or perhaps the function `getPodDeploymentsWithMetrics` is translating between the two. Looks like that might be the case. It's pretty confusing in there, but if it seems to be working then no need to clarify.